wallet/wallet.go

```diff
@@ -206,45 +207,66 @@ func CurrentAgendas(params *chaincfg.Params) (version uint32, agendas []chaincfg
 	return version, params.Deployments[version]
 }
 
-func (w *Wallet) readDBVoteBits(dbtx walletdb.ReadTx) stake.VoteBits {
-	version, deployments := CurrentAgendas(w.chainParams)
+func stakeVoteBits(version uint32) stake.VoteBits {
 	vb := stake.VoteBits{
 		Bits:         0x0001,
 		ExtendedBits: make([]byte, 4),
 	}
 	binary.LittleEndian.PutUint32(vb.ExtendedBits, version)
+	return vb
+}
+
+func (w *Wallet) readDBVoteBits(dbtx walletdb.ReadTx) (stake.VoteBits, map[string]stake.VoteBits) {
+	version, deployments := CurrentAgendas(w.chainParams)
+	defaultVb := stakeVoteBits(version)
+	ticketVbs := make(map[string]stake.VoteBits)
 
 	if len(deployments) == 0 {
-		return vb
+		return defaultVb, ticketVbs
 	}
 
 	for i := range deployments {
 		d := &deployments[i]
-		choiceID := udb.AgendaPreference(dbtx, version, d.Vote.Id)
-		if choiceID == "" {
+		defaultChoiceID, ticketsChoiceIDs := udb.AgendaPreferences(dbtx, version, d.Vote.Id)
```

Comment:
we should do the abstraction of whether to read the default, or the per-ticket vote preferences in the udb.AgendaPreference function call here.  I don't think this wallet func needs many/any changes to support this feature